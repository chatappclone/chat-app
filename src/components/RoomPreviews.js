import React from 'react';
import PropTypes from 'prop-types';
import RoomPreview from './RoomPreview';
import '../styles/RoomPreviews.scss';

function RoomPreviews({ rooms, roomMap, receiveHandleCurrentRoom, receiveNewChatView }) {
  return (
    <div>
      <nav className="room-previews__nav">
        <div className="room-previews__edit">Edit</div>
        <div>
          <h4>Chats</h4>
        </div>
        <div className="room-previews__new">
          <i
            className="zmdi zmdi-border-color"
            role="button"
            tabIndex={0}
            onClick={receiveNewChatView}
          />
        </div>
      </nav>
      <div className="room-previews__room-list">
        {rooms.map((room) => (
          <RoomPreview
            key={room.id}
            room={room}
            roomMap={roomMap}
            receiveHandleCurrentRoom={receiveHandleCurrentRoom}
          />
        ))}
      </div>
    </div>
  );
}

RoomPreviews.propTypes = {
  rooms: PropTypes.array.isRequired,
  roomMap: PropTypes.array.isRequired,
  receiveHandleCurrentRoom: PropTypes.func.isRequired,
  receiveNewChatView: PropTypes.func.isRequired,
};

export default RoomPreviews;
