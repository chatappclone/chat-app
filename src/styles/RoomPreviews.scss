@import 'defaults';

.room-previews {
  width: 100%;
  height: 100vh;
  font-size: 0.5rem;

  &__nav {
    background: #f7f7f7;
    width: 100%;
    height: 45px;
    display: flex;
    align-items: center;
    justify-content: center;
  }

  &__edit {
    color: #357af6;
    font-size: 0.85rem;
    padding-right: 127px;
  }

  &__title {
    font-size: 1rem;
    font-weight: 600;
  }

  &__room-list {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  &__new {
    color: #357af6;
    font-size: 0.85rem;
    padding-left: 138px;
  }
}
