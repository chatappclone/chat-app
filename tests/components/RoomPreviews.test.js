import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import RoomPreviews from '../../src/components/RoomPreviews';

const roomMap = [];
const rooms = [];
const receiveHandleCurrentRoom = jest.fn();
const receiveNewChatView = jest.fn();

describe('Room Previews', () => {
  test('matches the snapshot', () => {
    const tree = renderer
      .create(
        <RoomPreviews
          roomMap={roomMap}
          rooms={rooms}
          receiveHandleCurrentRoom={receiveHandleCurrentRoom}
          receiveNewChatView={receiveNewChatView}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('click open chat view', () => {
    const wrapper = shallow(
      <RoomPreviews
        roomMap={roomMap}
        rooms={rooms}
        receiveHandleCurrentRoom={receiveHandleCurrentRoom}
        receiveNewChatView={receiveNewChatView}
      />,
    );
    wrapper.find('.zmdi-border-color').simulate('click');
    expect(receiveNewChatView).toHaveBeenCalled();
  });
});
