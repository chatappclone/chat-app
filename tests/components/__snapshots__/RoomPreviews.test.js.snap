// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Room Previews matches the snapshot 1`] = `
<div
  className="room-previews"
>
  <nav
    className="room-previews__nav"
  >
    <div
      className="room-previews__edit"
    >
      Edit
    </div>
    <div
      className="room-previews__title"
    >
      Chats
    </div>
    <div
      className="room-previews__new"
    >
      <i
        className="zmdi zmdi-border-color"
        onClick={[MockFunction]}
        role="button"
        tabIndex={0}
      />
    </div>
  </nav>
  <div
    className="room-previews__room-list"
  />
</div>
`;
